### PR TITLE
Document testing local changes to lib-common

### PR DESCRIPTION
**Pull request by ASBishop**

### CODE REVIEW

File: testing_lib_common.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	github.com/openstack-k8s-operators/lib-common/modules/common v0.1.1-0.20231001084618-12369665b166
+	...
+)
+```
+
+At the bottom of each go.mod file, add a `replace` line that references your
+local lib-common repository. The reference may use an absolute path:
+
+```
+replace github.com/openstack-k8s-operators/lib-common/modules/common => /path/to/your/lib-common/modules/common
```
Comment (stuggi) [inline testing_lib_common.md:36]: could you please what you have to add in go.mod to be able to build your operator with a custom lib-common. This also allows e.g. to build/test an openstack-operator with a custom service operator

```
replace github.com/openstack-k8s-operators/lib-common/modules/common => github.com/<yourgithub>/lib-common/modules/common <commit hash>
```

running the `make tidy` will then replace the commit hash with the sudo version. after that you can build your operator with the custom lib-common as describe in https://github.com/openstack-k8s-operators/docs/blob/main/image_build.md

**Review by stuggi — COMMENTED** [review]
thanks a lot Alan for doing this doc! I was just asked yesterday on how to do it.

File: testing_lib_common.md (hunk 1 above)
Comment (ASBishop) [reply testing_lib_common.md:38]: Good idea, done!

**Review by stuggi — APPROVED** [review]
/lgtm thanks a lot!!

**Pull request merged**